runtime/src/iree/modules/vmvx/module.c

```diff
@@ -61,21 +61,431 @@ iree_vmvx_module_free_state(void* self, iree_vm_module_state_t* module_state) {
+                   /*size0=*/
+                   args->unary2d.size0,
+                   /*size1=*/
+                   args->unary2d.size1);
+
+  IREE_TRACE_ZONE_BEGIN(z0);
+  for (iree_device_size_t j = 0; j < out_size0; ++j) {
+    for (iree_device_size_t i = 0; i < out_size1; ++i) {
+      out[j * out_stride0 + i * out_stride1] =
+          in[j * in_stride0 + i * in_stride1];
+    }
+  }
+
+  IREE_TRACE_ZONE_END(z0);
+  return iree_ok_status();
+}
+
+IREE_VM_ABI_EXPORT(iree_vmvx_fill2d_x32, iree_vmvx_module_state_t, irIIII, v) {
+  IREE_TRACE_ZONE_BEGIN(z0);
+  MAP_BUFFER_2D_RW(out, int32_t,
+                   /*buffer_ref=*/
+                   args->fill2d_x32.out_ref,
+                   /*offset=*/
+                   args->fill2d_x32.out_offset,
+                   /*stride0=*/
+                   args->fill2d_x32.out_row_stride,
+                   /*stride1=*/
+                   1,
+                   /*size0=*/
+                   args->fill2d_x32.size0,
+                   /*size1=*/
+                   args->fill2d_x32.size1);
+
+  for (iree_device_size_t i = 0; i < out_size0; ++i) {
+    for (iree_device_size_t j = 0; j < out_size1; ++j) {
+      out[i * out_stride0 + j] = args->fill2d_x32.fill_value;
+    }
+  }
+
+  IREE_TRACE_ZONE_END(z0);
+  return iree_ok_status();
+}
+
+IREE_VM_ABI_EXPORT(iree_vmvx_matmul_f32f32f32, iree_vmvx_module_state_t,
+                   rIIrIIrIIIIIffi, v) {
+  IREE_TRACE_ZONE_BEGIN(z0);
+  MAP_BUFFER_2D_RO(lhs, float,
+                   /*buffer_ref=*/
+                   args->matmul_f32.lhs_ref,
+                   /*offset=*/
+                   args->matmul_f32.lhs_offset,
+                   /*stride0=*/
+                   args->matmul_f32.lhs_row_stride,
+                   /*stride1=*/
+                   1,
+                   /*size0=*/
+                   args->matmul_f32.m,
+                   /*size1=*/
+                   args->matmul_f32.k);
+  MAP_BUFFER_2D_RO(rhs, float,
+                   /*buffer_ref=*/
+                   args->matmul_f32.rhs_ref,
+                   /*offset=*/
+                   args->matmul_f32.rhs_offset,
+                   /*stride0=*/
+                   args->matmul_f32.rhs_row_stride,
+                   /*stride1=*/
+                   1,
+                   /*size0=*/
+                   args->matmul_f32.k,
+                   /*size1=*/
+                   args->matmul_f32.n);
+  MAP_BUFFER_2D_RW(out, float,
+                   /*buffer_ref=*/
+                   args->matmul_f32.out_ref,
+                   /*offset=*/
+                   args->matmul_f32.out_offset,
+                   /*stride0=*/
+                   args->matmul_f32.out_row_stride,
+                   /*stride1=*/
+                   1,
+                   /*size0=*/
+                   args->matmul_f32.m,
+                   /*size1=*/
+                   args->matmul_f32.n);
+
+  iree_host_size_t M = (iree_host_size_t)args->matmul_f32.m;
```

Comment:
iree_device_size_t

Reply:
Oops - this should be host_size_t (it is bounds checking that above -- here I am just doing it again for clarity (`M` -> `m`). But forgot to update the types on the loops. Doing that.